diesel/src/result.rs

```diff
@@ -156,6 +156,13 @@ pub trait DatabaseErrorInformation {
     /// Currently this method will return `None` for all backends other than
     /// PostgreSQL.
     fn constraint_name(&self) -> Option<&str>;
+
+    /// An optional integer indicating an error cursor position as an index into
+    /// the original statement string.
+    ///
+    /// Currently this method will return `None` for all backends other than
+    /// PostgreSQL.
+    fn statement_position(&self) -> Option<f32>;
```

Comment:
Is there any reason to return a floating point number here?

Reply:
From the [docs](https://www.postgresql.org/docs/13/libpq-exec.html):

> A string containing a decimal integer indicating an error cursor position as an index into the original statement string. The first character has index 1, and positions are measured in characters not bytes.

As it's a decimal, floating-point numbers seemed the most reasonable.

Reply:
It also says it's an integer and that positions are measured in characters so I cannot see how this wouldn't be a integer.